build_tools/cmake/iree_copts.cmake

```diff
@@ -294,7 +294,7 @@ iree_select_compiler_opts(IREE_DEFAULT_COPTS
 #   https://discourse.cmake.org/t/how-to-fix-build-warning-d9025-overriding-gr-with-gr/878
 #   https://gitlab.kitware.com/cmake/cmake/-/issues/20610
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
-  string(REPLACE "/GR" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+  string(REPLACE "/GR" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```

Comment:
Ugh CMake. I can't stand this "sometimes actually empty, sometimes an empty string representing an empty list" typing...

We do similar replacements in other places, but for variables that we initialize ourselves.

With your changes, this has the correct behavior, but we could instead explicitly check for `CMAKE_CXX_FLAGS` being defined and only modify it if it is.

```
if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC" AND CMAKE_CXX_FLAGS)
  string(REPLACE "/GR" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
endif()
```

Reply:
Good points. Modified.